src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -0,0 +1,618 @@
+
++---------------+---------------+-------------------+
+|               |   C++ Class   |  C Type (Handle)  |
++===============+===============+===================+
+|   Device      | xrt::device   |  xrtDeviceHandle  |
++---------------+---------------+-------------------+
+|   XCLBIN      | xrt::xclbin   |  xrtXclbinHandle  |
++---------------+---------------+-------------------+
+|   Buffer      | xrt::bo       |  xrtBufferHandle  |
++---------------+---------------+-------------------+
+|   Kernel      | xrt::kernel   |  xrtKernelHandle  |
++---------------+---------------+-------------------+
+|   Run         | xrt::run      |  xrtRunHandle     |
++---------------+---------------+-------------------+
+
+The common host code flow using the above data structures is as below
+   
+      - Open Xilinx **Device** and Load the **XCLBIN**
+      - Set up the **Buffers** that are used to transfer the data between the host and the device
+      - Use the Buffer APIs for the data transfer between host and device (before and after the kernel execution).
+      - Use **Kernel** and **Run** handle/objects to offload and manage the compute-intensive tasks running on FPGA. 
+       
+      
+Below we will walk through the common API usage to accomplish the above tasks. 
+
+Device and XCLBIN
+-----------------
+
+Device and XCLBIN class provide fundamental infrastructure-related interfaces. The primary objective of the device and XCLBIN related APIs are
+ 
+    - Open a Device
+    - Load compiled kernel binary (or XCLBIN) onto the device 
+
+
+Example C API based code  
+
+.. code:: c
+      :number-lines: 10
+           
+           xrtDeviceHandle device = xrtDeviceOpen(0);
+       
+           xrtXclbinHandle xclbin = xrtXclbinAllocFilename("kernel.xclbin");
+       
+           xrtDeviceLoadXclbinHandle(device,xclbin);
+           ..............
+           ..............
+           xrtDeviceClose(device);
+
+       
+
+The above code block shows
+      
+      - Opening the device (enumerated as 0) and get device handle ``xrtDeviceHandle`` (line 10)
+          
+          - Device indices are enumerated as 0,1,2 and can be observed by ``xbutil scan``
+          
+          .. code::
+               
+               >>xbutil scan
+               INFO: Found total 2 card(s), 2 are usable
+               .............
+               [0] 0000:b3:00.1 xilinx_u250_gen3x16_base_1 user(inst=129)
+               [1] 0000:65:00.1 xilinx_u50_gen3x16_base_1 user(inst=128)
+
+      - Opening the XCLBIN from the filename and get an XCLBIN handle ``xrtXclbinHandle`` (line 12)
+      - Loading the XCLBIN onto the Device by using the XCLBIN handle by API ``xrtDeviceLoadXclbinHandle`` (line 14)
+      - Closing the device handle at the end of the application (line 19)
+      
+
+**C++**: The equivalent C++ API based code
+
+.. code:: c++
+      :number-lines: 10
+           
+           auto device = xrt::device(0);
```

Comment:
My preference.  I think it much clearer coding to have the types on the right (pun intended) side.  I don't like your suggestion.

Reply:
nothing is needed to do for this PR, resolving

Reply:
My very problem here is that I am incompetent in your API (I am sorry about this, but I guess most of the readers will be in that casetoo...), so it is not clear whether `xrt::device` is some factory function or a `device` constructor.
To follow your coding style,
```c++
auto device = xrt::device { 0 };
```
make clear that `xrt::device` is a type in your API because you cannot call a function this way.

Reply:
Yeah, that's a good point about using list initialization.  I have never really gotten used to that, but clearly that is what should be used in this case.

Reply:
> Yeah, that's a good point about using list initialization. I have never really gotten used to that, but clearly that is what should be used in this case.

Really? It comes from old C... ;-)